TASVideos/TagHelpers/Buttons/AddLinkTagHelper.cs

```diff
@@ -6,10 +6,13 @@ namespace TASVideos.TagHelpers;
 
 public class AddLinkTagHelper(IHtmlGenerator generator) : AnchorTagHelper(generator)
 {
+	public override void Process(TagHelperContext context, TagHelperOutput output)
+		=> ProcessAsync(context, output).Wait();
```

Comment:
This is bad. Why are we doing this? .Wait() on async calls is asking for deadlocks

Reply:
The first commit is to fix an inconsistency in subclasses of `AnchorTagHelper`. [In `TagHelper`, `ProcessAsync` calls `Process`](https://github.com/dotnet/aspnetcore/blob/a813ad450371e4c72578ac2d2d39a57e49dad23a/src/Razor/Razor/src/TagHelpers/TagHelper.cs#L39-L55), which is fine. `AnchorTagHelper` overrides `Process` *only*, which is fine. But those subclasses were overriding `ProcessAsync` only, so if `Process` is called (which I guess was never happening, thankfully) then it would be the implementation from `AnchorTagHelper`, which is obviously not going to produce the same output.

Reply:
Can we not implement both Process and ProcessAsync? The .Wait() is a no go for me, need another way